anime_downloader/sites/helpers/selescrape.py

```diff
@@ -0,0 +1,191 @@
+        chrome_options = Options()
+        chrome_options.add_argument("--headless")
+        chrome_options.add_argument("--disable-gpu")
+        profile_path = os.path.join(data_dir, 'Selenium_chromium')
+        log_path = os.path.join(data_dir, 'chromedriver.log')
+        chrome_options.add_argument(f'--log-path {log_path}')
+        chrome_options.add_argument(f"--user-data-dir={profile_path}")
+        chrome_options.add_argument("--no-sandbox")
+        chrome_options.add_argument("--window-size=1920,1080")
+        chrome_options.add_argument(f'user-agent={get_random_header()}')
+        if driver_binary == '.':
+            if executable == '.':
+                driver = webdriver.Chrome(options=chrome_options)
+            else:
+                from selenium.webdriver.common.desired_capabilities import DesiredCapabilities
+                cap = DesiredCapabilities.CHROME
+                cap['binary_location'] = executable
+                driver = webdriver.Chrome(desired_capabilities=cap, options=chrome_options)
+        else:
+            if executable == '.':
+                driver = webdriver.Chrome(options=chrome_options)
+            else:
+                from selenium.webdriver.common.desired_capabilities import DesiredCapabilities
+                cap = DesiredCapabilities.CHROME
+                cap['binary_location'] = executable
+                driver = webdriver.Chrome(executable_path=driver_binary, desired_capabilities=cap, options=chrome_options)
+    return driver
+
+
+def status_select(driver, url, status='hide'):
+    """
+    for now it doesnt do what its name suggests, 
+    i have planned to add a status reporter of the http response code.
+    """
+    try:
+        if status == 'hide':
+            driver.get(url)
+        elif status == 'show':
+            r = requests.head(url)
+            if r.status_code == 503:
+                raise exception("WARNING: This website's sevice is unavailable or has cloudflare on.")
+            driver.get(url)
+            return r.status_code
+        else:
+            driver.get(url)
+    except requests.ConnectionError:
+        raise exception("Failed to establish a connection using the requests library.")
+
+        
+def cloudflare_wait(driver):
+    """
+    it waits until cloudflare has gone away before doing any further actions.
+    """
+    abort_after = 30
+    start = time.time()
+
+    title = driver.title  # title = "Just a moment..."
+    while title == "Just a moment...":
+        time.sleep(0.25)
+        delta = time.time() - start
+        if delta >= abort_after:
+            raise Exception(f'Timeout:\nCouldnt bypass cloudflare. See the screenshot for more info:\n{get_data_dir()}/screenshot.png')
+        title = driver.title
+        if not title == "Just a moment...":
+            break
+    time.sleep(1)
+    
+    
+def request(url, request_type='GET', **kwargs): #Headers not yet supported , headers={}
```

Comment:
you've switched places with url and request-type, looking at request.py it passes the request-type first. Something like `def request(request_type, url, **kwargs): """Headers not yet supported , headers={}"""` works.